src/Grpc.Net.Client/Internal/StreamExtensions.cs

```diff
@@ -62,6 +62,13 @@ private static Status CreateUnknownMessageEncodingMessageStatus(string unsupport
             GrpcCallLog.ReadingMessage(call.Logger);
             cancellationToken.ThrowIfCancellationRequested();
 
+#if NET7_0_OR_GREATER
```

Comment:
Why the ifdef? Something to do with the underlying stream impl?

Reply:
Yes. Some HttpHandler stream implementations in old frameworks don't like zero byte buffers, e.g. `Microsoft.AspNetCore.TestHost.ResponseBodyReaderStream`